### PR TITLE
tidb-in-kubernetes: add default cluster requirements

### PR DESCRIPTION
**Pull request by juliezhang1112**
### What is changed, added or deleted? 
- Add default cluster requirements: storage class and deployment topology
These two aspects usually lead to deployment failures if users are unaware of it.
- Add a note about multiple clusters management

### What is the related PR or file link(s)? 
https://github.com/pingcap/docs-cn/pull/1778

### Which version does your change affect? 
dev, v3.0

### CODE REVIEW

**Comment by lilin90** [comment]
@juliezhang1112 Please resolve conflicts.

**Comment by juliezhang1112** [comment]
PTAL @TomShawn

**Comment by juliezhang1112** [comment]
PTAL @lilin90 @TomShawn @tennix

**Comment by TomShawn** [comment]
LGTM

File: dev/tidb-in-kubernetes/deploy/general-kubernetes.md [hunk 1]
```diff
@@ -31,7 +31,30 @@ helm inspect values pingcap/tidb-cluster --version=<chart-version> > /home/tidb/
...
+The TiDB cluster uses `local-storage` by default.
+
+- For the production environment, local storage is recommended. The actual local storage in Kubernetes clusters might be sorted by disk types, such as `nvme-disks` and `sas-disks`.
+- For the demonstration environment or functional verification, you can use network storage, such as `ebs` and `nfs`.
+
+Different components of a TiDB cluster have different disk requirements. Before deploying a TiDB cluster, select the appropriate storage class for each component according to the storage classes supported by the current Kubernetes cluster and usage scenario. You can set the storage class by modifying `storageClassName` of each component in `values.yaml`. For the [storage classes](/dev/tidb-in-kubernetes/reference/configuration/local-pv.md) supported by the Kubernetes cluster, check with your system administrator.
+
+> **Note:**
+>
+> If you set a storage class that does not exist in the TiDB cluster that you are creating, then the cluster creation is in the Pending state. In this situation, you must [destroy the TiDB cluster in Kubernetes](/dev/tidb-in-kubernetes/maintain/destroy-tidb-cluster.md).
```
Comment (lilin90) [inline dev/tidb-in-kubernetes/deploy/general-kubernetes.md]:
```suggestion
> If you set a storage class that does not exist in the TiDB cluster that you are creating, then the cluster creation goes to the Pending state. In this situation, you must [destroy the TiDB cluster in Kubernetes](/dev/tidb-in-kubernetes/maintain/destroy-tidb-cluster.md).
```

File: dev/tidb-in-kubernetes/deploy/general-kubernetes.md [hunk 2]
```diff
@@ -31,7 +31,30 @@ helm inspect values pingcap/tidb-cluster --version=<chart-version> > /home/tidb/
...
+
+### Cluster topology
+
+The deployed cluster topology by default has 3 PD Pods, 3 TiKV Pods, 2 TiDB Pods, and 1 Monitor Pod. In this deployment topology, the scheduler extender of TiDB Operator requires at least 3 nodes in the Kubernetes cluster based on the principle of high availability. If the number of Kubernetes cluster nodes is less than 3, 1 PD Pod is in the Pending state, and neither TiKV Pods nor TiDB Pods are created.
```
Comment (lilin90) [inline dev/tidb-in-kubernetes/deploy/general-kubernetes.md]:
```suggestion
The deployed cluster topology by default has 3 PD Pods, 3 TiKV Pods, 2 TiDB Pods, and 1 Monitor Pod. In this deployment topology, the scheduler extender of TiDB Operator requires at least 3 nodes in the Kubernetes cluster to provide high availability. If the number of Kubernetes cluster nodes is less than 3, 1 PD Pod goes to the Pending state, and neither TiKV Pods nor TiDB Pods are created.
```

**Pull request merged**